py-polars/polars/utils/udfs.py

```diff
@@ -178,6 +182,35 @@ def _inject_nesting(
 
         return sorted(expression_blocks.items())
 
+    def _get_target_name(self, col: str, expression: str) -> str:
+        """The name of the object against which the 'apply' is being invoked."""
+        if self._apply_target_name is not None:
+            return self._apply_target_name
+        else:
+            col_expr = f'pl.col("{col}")'
+            if self._apply_target == "expr":
+                return col_expr
+            elif self._apply_target == "series":
+                # handle overlapping name from global variables; fallback through
+                # "s", "srs", "series", and then the series' name (if applicable)
+                series_aliases = ["s", "srs", "series"]
+                if col.isalpha() and col not in series_aliases:
+                    series_aliases.append(col)
```

Comment:
why is this necessary?

Reply:
Should actually be `.isidentifier` rather than `isalpha` (I keep forgetting that exists - will update).

Reply:
even so, not sure I sure why it's necessary at all. I tried deleting it and the tests passed - any chance you could expand on this please (or, even better, add a test for which it's necessary)?

Reply:
It's necessary if the series name has a space, starts with a number, or contains non-alphanumeric chars ;)

Reply:
ah, right!
```python
In [3]: pl.Series('bob', [1,2,3]).apply(lambda x: x+s+srs+series)
<ipython-input-3-e199d2d5c974>:1: PolarsInefficientApplyWarning: 
Series.apply is significantly slower than the native series API.
Only use if you absolutely CANNOT implement your logic otherwise.
In this case, you can replace your `apply` with the following:
  -  bob.apply(lambda x: ...)
  +  ((bob + s) + srs) + series
```

How about we just delete this and go straight to `srs0`? it's already quite a lot to parse while reading :) 

final comment here, will approve after this one 😇

Reply:
> How about we just delete this and go straight to `srs0`? it's already quite a lot to parse while reading :)

Done; the chance of ever hitting it is already negligible - let's simplify!